src/compiler/diagnosticMessages.json

```diff
@@ -3907,6 +3907,18 @@
         "category": "Error",
         "code": 2868
     },
+    "This expression is always truthy. Did you mean to test something else?": {

```

Comment:
nit: I don't like using 'Did you mean' unless we have something specific to suggest.
I'd prefer either just "This expression is always truthy" or "Using XX on this expression appears unintentional because it always evaluates the left-hand side".

Probably the former, because the squiggly should be enough to point out the error to people -- more than a longer message, at least.